package/spec/frontend/media_player/hooks_spec.js

```diff
@@ -64,27 +62,31 @@ describe('hooks', function() {
         expect(player.intendingToPause()).toBe(false);
       });
 
-      it('calls original play method when promise returned by before is resolved', function() {
+      it('calls original play method when promise returned by before is resolved', async () => {
         var player = fakePlayer();
-        var deferred = new jQuery.Deferred();
+
+        var promiseResolve;
+        var promise = new Promise(function(resolve, reject) {
+          promiseResolve = resolve;
+        });
+
         hooks(player, {
           before: function() {
-            return deferred.promise();
+            return promise;
           }
         });
 
         player.play();
-        deferred.resolve();
+        await promiseResolve();
```

Comment:
Awaiting resolve does nothing since it does not return a promise. We instead have to await the play. We cannot await `play` directly, since it stays pending until the `before` promise is resolved. We could do

```
const playPromise = player.play();
promiseResolve()
await playPromise
```